### ISSUE TITLE
How to reproduce the results of Induction Head task?

### ISSUE THREAD

**Issue by Karami-m** [state-spaces/mamba#298]
This work is very inspiring. I have a specific questions about induction head task. 

1. Can you please provide more details about the **model and experiments configuration** that you used for this experiments? Based on the information presented in the paper, I configured the following parameters for my experiments:

```
dataset:
  vocab_size: 16
  input_seq_len: 256
  batch_size: 8
  
optimizer:
  lr: 1e-3
  weight_decay: 0.1

model:
  config:
    d_model: 64
    n_layer: 2
    ssm_cfg:
      d_state: 16
      d_conv: 4
      expand: 2
```

I experimented with `weight_decay= [0.001, 0.1] `, but the model failed to learn the task and train/test accuracy fluctuated at around 7%. However, it was able to learn the other task of [associative recall](https://github.com/HazyResearch/safari/tree/main/configs/experiment/synthetics/associative_recall) with such setting.

For the codebase, I integrated Mamba as a sequence mixer into the [Hyena codebase](https://github.com/HazyResearch/safari) for [Induction Head Task](https://github.com/HazyResearch/safari/tree/main/configs/experiment/synthetics/induction_head).

2. **Training loss**: from the implementation  of such model as an auto-regressive the train loss is evaluated on all tokens and test is only evaluated on the last token. I also tried to use the representation of the last token, by simply modelling the training task as a  **classification problem**, i.e. loss and accuracy of training and test are evaluated only on the last token as such that 
$L( {y}[..., -1]),  x[..., -1])$
